### PR TITLE
Add a tracing policy to enable fine grained tracing per component con…

### PR DESCRIPTION
**Pull request by vietj**
…figuration.

- https://github.com/eclipse-vertx/vertx-sql-client/pull/793
- https://github.com/eclipse-vertx/vertx-tracing/pull/10

### CODE REVIEW

**Comment by vietj** [comment]
ping @jotak

File: src/main/java/io/vertx/core/eventbus/impl/HandlerRegistration.java [hunk 1]
```diff
@@ -155,10 +156,11 @@ public void next() {
           bus.metrics.messageDelivered(m, message.isLocal());
         }
         if (tracer != null && !src) {
-          message.trace = tracer.receiveRequest(context, message, message.isSend() ? "send" : "publish", message.headers(), MessageTagExtractor.INSTANCE);
+          message.trace = tracer.receiveRequest(context, TracingPolicy.PROPAGATE, message, message.isSend() ? "send" : "publish", message.headers(), MessageTagExtractor.INSTANCE);
```
Comment (jotak) [inline src/main/java/io/vertx/core/eventbus/impl/HandlerRegistration.java:159]: no configurable policies for the event bus? is that on purpose?

File: src/main/java/io/vertx/core/eventbus/impl/ReplyHandler.java [hunk 2]
```diff
@@ -45,7 +45,8 @@
 
   private void trace(Object reply, Throwable failure) {
     VertxTracer tracer = context.tracer();
-    if (tracer != null && src) {
+    Object trace = this.trace;
```
Comment (jotak) [inline src/main/java/io/vertx/core/eventbus/impl/ReplyHandler.java:48]: why this new Object? thread-safety concerns?

File: src/main/java/io/vertx/core/tracing/TracingPolicy.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+  /**
+   * Propagate an existing trace.
+   */
+  PROPAGATE,
+
+  /**
+   * Reuse an existing trace or create a new trace when no one exist.
+   */
+  REQUIRED
```
Comment (jotak) [inline src/main/java/io/vertx/core/tracing/TracingPolicy.java:33]: as said in the other PR I think I would prefer "Always" 
or "CREATE_OR_PROPAGATE" would be very explicit but very verbose as well...

File: src/main/java/io/vertx/core/eventbus/impl/ReplyHandler.java (hunk 2 above)
Comment (vietj) [reply src/main/java/io/vertx/core/eventbus/impl/ReplyHandler.java:48]: yes, reading a ref and then using it

File: src/main/java/io/vertx/core/tracing/TracingPolicy.java (hunk 3 above)
Comment (vietj) [reply src/main/java/io/vertx/core/tracing/TracingPolicy.java:33]: I'll use always as you recommend

**Review by jotak — APPROVED** [review]

**Pull request merged**